### DISCUSSION TITLE
making line art from image

### DISCUSSION THREAD

**Discussion by abilask** [ImageMagick/ImageMagick#5254] [Help]
Hi , I would like to get a sample program for making line art from image, example given "input.jpg" and the expected out put image is "output.png", my environment is  ImageMagick-7.1.0-30 version on Linux (Ubuntu). Thanks in advance.
![input](https://user-images.githubusercontent.com/90464818/174840353-1b26f261-9276-4aa0-a14c-40fbb7353eba.jpg)
![output](https://user-images.githubusercontent.com/90464818/174840364-c66e0716-307c-4f44-ae66-9c58d1066f13.png)

**Comment by fmw42** [comment]
This will be hard because of the texture and reflections on your background.  Can you photograph on a less textured and less reflection background?

`magick bunny.jpg -canny 0x1+1%+30% x.png
`
![x](https://user-images.githubusercontent.com/13458218/174843598-f615f11a-a665-43ab-a7ec-b826b7cb7564.png)

**Reply by abilask** [reply]
is this image is fine?
![863cee4e29cc9747ec1d11683cdf3268](https://user-images.githubusercontent.com/90464818/174918944-f721eb85-40ad-4a19-b3cf-62dc6c21929c.jpg)

**Reply by fmw42** [reply]
Not really -- it has dark shadows

**Reply by abilask** [reply]
this ?
![tangram](https://user-images.githubusercontent.com/90464818/174932300-cc2b664e-fd04-4f31-9a8e-cb89e08a7f23.gif)

**Reply by fmw42** [reply]
`magick origami.gif -canny 0x1+10%+30% x.png`

![x](https://user-images.githubusercontent.com/13458218/174937143-348ca881-b691-4399-ae79-369142ec7948.png)

**Comment by snibgo** [comment]
As fmw42 says.

The best background is one that is visibly different to the foreground, for example a strong and fairly constant colour. Then the background won't have edges, and you can easily make it transparent if you want.
